spec/20-BINARY_FORMAT.md

```diff
@@ -1,3 +1,125 @@
 # Binary format
 
-TBD
\ No newline at end of file
+Binary format document describes how to encode each field - `traceparent` and
+`tracestate`. The binary format should be used to encode the values of these
+fields. This specification does not specify how these fields should be stored
+and send as a part of a binary payload. The basic implementation may serialize
+those as size of the field followed by the value.
+
+# `Traceparent` binary format
+
+The field `traceparent` encodes the version of the protocol and fields
+`trace-id`, `parent-id` and `recorded`. Each field starts with the one byte
+field identifier with the field value following immediately after it. Field
+identifiers are used as markers for additional verification of the value
+consistency and may be used in future for the versioning of the `traceparent`
+field.
+
+``` abnf
+traceparent     = version version_format  
+version         = 1BYTE                   ; version is 0 in the current spec
+version_format  = "{ 0x0 }" trace-id "{ 0x1 }" parent-id "{ 0x2 }" recorded
+trace-id        = 16BYTES
+parent-id       = 8BYTES
+trace-flags     = 1BYTE  ; only the least significant bit is used
+```
+
+Unknown field identifier (anything beyond `0`, `1` and `2`) should be treated as
+invalid `traceparent`. All zeroes in `trace-id` and `parent-id` invalidates the
+`traceparent` as well.
+
+## Serialization of `traceparent`
+
+Implementation MUST serialize fields into the field ordering sequence.
+In other words, `trace-id` field should be serialized first, `parent-id`
+second and `trace-flags` - third.
+
+Field identifiers should be treated as unsigned byte numbers and should be
+encoded in big-endian bit order.
+
+When `trace-id` is represented as a byte array - first element of an array MUST
+be copied first. When array is represented as a memory block of 16 bytes -
+serialization of `trace-id` would be identical to `memcpy` method call on that
+memory block. Same applies to `parent-id` field.
+
+If padding of the field is required (`traceparent` needs to be serialized into
+the bigger buffer) - any number of bytes can be appended to the end of the
+serialized value.
+
+## De-serialization of `traceparent`
```

Comment:
I am not sure there's much value in this section

Reply:
feedback for the http spec was that algorithm of parsing would be very helpful for understanding the spec. Is your concern that this should be non-normative as we must not prescribe the specific algorithm? I can move it to non-normative section, it may be a good idea.

Reply:
I think code / reference implementation, rather than narrative, is the best way to describe the algorithm  :-)

Reply:
One thing this algo has that is missing in a spec is how to read a byte: "Field identifier should be read as unsigned byte assuming big-endian bits order"

SO your opinion that this algo absolutely out of scope of this spec? Or you just want to move it out of normative section?

Reply:
Moved to the separate non-formative section and made comment about bytes